### PR TITLE
treewide: Fix some unstableGitUpdater users

### PR DESCRIPTION
**Pull request by OPNA2608**
## Description of changes

#280501 now makes `unstableGitUpdater` produce versions that match our desired unstable version schema. Start fixing the versions of packages that use it, and their `unstableGitUpdater` arguments as needed.

Tackling abunch of mostly self-contained packages here. The only directory left after this from the initial treewide PR is `pkgs/development/libraries`, where I expect the amount of rebuilds to shoot up abit. Checked running the update script of everything, which results in either the same version & revs as corrected here, or newer versions.

I *think* the `klipper-*` failures are also present on [Hydra](https://hydra.nixos.org/eval/1806092?filter=klipper&compare=1806084), though the only information there is that the output limit was exceeded? Failure from the review reproduces locally on master at least: [klipper-firmware-mcu-0.12.0-unstable-2024-04-20.log](https://github.com/NixOS/nixpkgs/files/15213746/klipper-firmware-mcu-0.12.0-unstable-2024-04-20.log)

(If you wish to run the review locally, then I *really* recommend excluding `open-watcom-v2*` from it. It's a very slow, single-core build with a very low impact. As its package maintainer feel free to not waste your time on it :slightly_smiling_face:)

---

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>c0</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>klipper-firmware</li>
    <li>klipper-flash</li>
  </ul>
</details>
<details>
  <summary>31 packages built:</summary>
  <ul>
    <li>bloat</li>
    <li>dracula-theme</li>
    <li>femtolisp</li>
    <li>heroic</li>
    <li>heroic-unwrapped</li>
    <li>klipper</li>
    <li>material-kwin-decoration</li>
    <li>moonraker</li>
    <li>nelua</li>
    <li>nile</li>
    <li>nile.dist</li>
    <li>nixos-bgrt-plymouth</li>
    <li>nushellPlugins.net</li>
    <li>pantheon.file-roller-contract</li>
    <li>postgresql12JitPackages.pgjwt (postgresql13JitPackages.pgjwt ,postgresql14JitPackages.pgjwt ,postgresql15JitPackages.pgjwt ,postgresql16JitPackages.pgjwt)</li>
    <li>postgresql12JitPackages.tds_fdw</li>
    <li>postgresql12Packages.pgjwt (postgresql13Packages.pgjwt ,postgresql14Packages.pgjwt ,postgresql15Packages.pgjwt ,postgresql16Packages.pgjwt)</li>
    <li>postgresql12Packages.tds_fdw</li>
    <li>postgresql13JitPackages.tds_fdw</li>
    <li>postgresql13Packages.tds_fdw</li>
    <li>postgresql14JitPackages.tds_fdw</li>
    <li>postgresql14Packages.tds_fdw</li>
    <li>postgresql15JitPackages.tds_fdw</li>
    <li>postgresql15Packages.tds_fdw</li>
    <li>postgresql16JitPackages.tds_fdw</li>
    <li>postgresql16Packages.tds_fdw</li>
    <li>rc-9front</li>
    <li>scheme-manpages</li>
    <li>wingpanel-indicator-ayatana</li>
    <li>zig-shell-completions</li>
    <li>zsh-prezto</li>
  </ul>
</details>

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/desktops/pantheon/third-party/wingpanel-indicator-ayatana/default.nix
Comment (eclairevoyant) [inline pkgs/desktops/pantheon/third-party/wingpanel-indicator-ayatana/default.nix:1]: note (unrelated to PR): is abandoned and superceded by https://github.com/MvBonin/wingpanel-community-indicators

File: pkgs/games/doom-ports/slade/git.nix
Comment (eclairevoyant) [inline pkgs/games/doom-ports/slade/git.nix:1]: This one could use a version bump, not sure why the updater script wouldn't catch this?

File: pkgs/servers/klipper/default.nix [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 stdenv.mkDerivation rec {
   pname = "klipper";
-  version = "unstable-2024-04-20";
+  version = "0.12.0-unstable-2024-04-20";
 
   src = fetchFromGitHub {
     owner = "KevinOConnor";
```
Comment (eclairevoyant) [inline pkgs/servers/klipper/default.nix:14]: note: repo relocated to https://github.com/Klipper3d/klipper

**Review by eclairevoyant — COMMENTED** [review]
only feedback here is really for `sladeUnstable`, rest seem fine

File: pkgs/games/doom-ports/slade/git.nix
Comment (OPNA2608) [reply pkgs/games/doom-ports/slade/git.nix:1]: https://github.com/NixOS/nixpkgs/pull/278799, apparently waiting for the package maintainer to resolve an issue with it.

**Pull request merged**